CONTRIBUTING.md

```diff
@@ -0,0 +1,271 @@
+of the same idea? Are there relevant issues or discussions elsewhere? If
+any of these questions provides valuable information, answer it.
+Otherwise, feel free to leave it out—some changes really are
+self-documenting, and there’s no need to add a vacuous description.
+
+<details>
+<summary>Why include a description?</summary>
+
+A commit describes a _change_ from one state of the codebase to the
+next. If your patch is good, the final state of the code will be clear
+to anyone reading it. But this isn’t always sufficient to explain why
+the change was necessary. Documenting the motivation, alternate
+formulations, etc. is helpful both in the present (for reviewers) and in
+the future (for people using `git-blame` to try to understand how a
+piece of code came to be).
+</details>
+
+### Test plan
+
+After the description, include a **test plan**. Describe what someone
+should do to verify that your changes are correct. This can include
+automated tests, manual tests, or tests of the form “verify that when
+you change the code in this way, you see this effect.” Feel free to
+include shell commands and expected outputs if helpful.
+
+Sometimes, the test plan may appear trivial. It may be the case that you
+only ran the standard unit tests, or that you didn’t feel that any
+testing at all was necessary. In these cases, you should still include
+the test plan: this signals to observers that the trivial steps are
+indeed sufficient.
+
+<details>
+<summary>Why include a test plan?</summary>
+
+The value of a test plan is many-fold. Simply writing the test plan can
+force you to consider cases that you hadn’t before, in turn helping you
+discover bugs or think of alternate implementations. Even if the test
+plan is as simple as “standard unit tests suffice”, this indicates to
+observers that no additional testing is required. The test plan is
+useful for reviewers, and for anyone bisecting through the history or
+trying to learn more about the development or intention of a commit.
+</details>
+
+### Wrapping
+
+Wrap all parts of the commit message so that no line has more than **72
+characters**.
+
+<details>
+<summary>Why wrap at 72 characters?</summary>
+
+This leaves room for four spaces of padding on either side while still
+fitting in an 80-character terminal. Programs like `git-log` expect that
+this amount of padding exists.
+
+(Yes, people really still use 80-character terminals. When each of your
+terminals has bounded width, you can display more of them on a screen!)
+</details>
+
+### Example
+
+Here is an example of a helpful commit message. [The commit in
+question][example-commit] doesn’t change very many lines of code, but
+the commit message explains the context behind the commit, links to
+relevant issues, thanks people who contributed to the commit, and
+describes a manual test plan. Someone reading this commit for the first
+time will have a much better understanding of the change by reading this
+commit message:
+
+```
```

Comment:
Also happy to format this as a block-quote instead of code.

Reply:
I looked into this a bit, and prefer it as is. Formatting as a block
quote changes the wrapping, the list formatting, and the inline code
formatting, making it harder for readers to understand what was actually
written without looking at the source.